### DISCUSSION TITLE
Changing nextConfig to fix an issue with aws-crt is not working with pwa export .

### DISCUSSION THREAD

**Discussion by danalex7212** [DuCanhGH/next-pwa#76] [Q&A]
I changed  next.config.ts to address an issue regarding a warning using aws-crt outlined in this comment 
https://github.com/aws-amplify/amplify-js/issues/11030#issuecomment-1598207365

My next.config.js file now is 

```
// @ts-check
/** @type {import('next').NextConfig} */

const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  modularizeImports: {
    "@mui/icons-material": {
      transform: "@mui/icons-material/{{member}}",
    },
  },
  webpack: (config, { webpack, isServer, nextRuntime }) => {
    // Avoid AWS SDK Node.js require issue
    if (isServer && nextRuntime === "nodejs")
      config.plugins.push(
        new webpack.IgnorePlugin({
          resourceRegExp: /^aws-crt$/,
        })
      );
    return config;
  },
};

const withPWA = require("@ducanh2912/next-pwa").default({
  dest: "public",
});

module.exports = withPWA({ nextConfig });
```

If I export nextConfig without withPWA the issue is fixed according to the comment linked above . 
But when I export using withPWA its still not resolved . Why ? 
My warning according to issue of the comment linked above .

![image](https://github.com/DuCanhGH/next-pwa/assets/36897394/e329c034-a9ef-443f-9405-d1c23f1dc011)

**Comment by DuCanhGH** [comment]
You are supposed to pass `nextConfig` to `withPWA` like this: `withPWA(nextConfig);`. Not sure why your IDE didn't warn you.

**Reply by danalex7212** [reply]
Thank you .
